static/app/components/input.tsx

```diff
@@ -1,21 +1,87 @@
+import {forwardRef} from 'react';
 import isPropValid from '@emotion/is-prop-valid';
+import {css} from '@emotion/react';
 import styled from '@emotion/styled';
 
-import {inputStyles} from 'sentry/styles/input';
+import {FormSize, Theme} from 'sentry/utils/theme';
 
 export interface InputProps
-  extends React.InputHTMLAttributes<HTMLInputElement>,
-    Omit<Parameters<typeof inputStyles>[0], 'theme'> {
+  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size'> {
+  monospace?: boolean;
+  nativeSize?: React.InputHTMLAttributes<HTMLInputElement>['size'];
+  size?: FormSize;
   type?: React.HTMLInputTypeAttribute;
 }
 
-const Input = styled('input', {
-  shouldForwardProp: prop =>
-    // Do not forward required to `input` to avoid default browser behavior
-    typeof prop === 'string' && isPropValid(prop) && prop !== 'required',
-})<InputProps>`
-  ${inputStyles};
+export const inputStyles = (p: InputProps & {theme: Theme}) => css`
```

Comment:
So it looks like we've got `inputStyles` being exported from two places now - here, and in `sentry/styles/input`. Are we okay with having both exist?

Reply:
Nope! The `inputStyles` exported here will the the main one. My plan is to go through all the `sentry/styles/input` imports and either replace them with this `<Input />` component, or if that's not possible redirect the import path to here (`sentry/components/input`). After all that is done, I'll delete `styles/input`. This PR is the first step in that process.

Reply:
Sounds good! 👍